src/main/java/com/minecolonies/entity/ai/citizen/miner/EntityAIStructureMiner.java

```diff
@@ -165,21 +167,30 @@ private String getRenderMetaTorch()
     }
 
     /**
-     * Can be overridden by implementations to specify items useful for the worker.
-     * When the workers inventory is full, he will try to keep these items.
-     * ItemStack amounts are ignored, the first stack found will be taken.
+     * Override this method if you want to keep an amount of items in inventory.
+     * When the inventory is full, everything get's dumped into the building chest.
+     * But you can use this method to hold some stacks back.
      *
-     * @return a list with items nice to have for the worker
+     * @return a list of objects which should be kept.
      */
-    @NotNull
-    @Override
-    protected List<ItemStack> itemsNiceToHave()
+    protected Map<ItemStorage, Integer> needXForWorker()
     {
-        return Arrays.asList(new ItemStack(Blocks.LADDER),
-          new ItemStack(Blocks.PLANKS),
-          new ItemStack(Blocks.OAK_FENCE),
-          new ItemStack(Blocks.TORCH),
-          new ItemStack(Blocks.COBBLESTONE));
+        Map<ItemStorage, Integer> keepX = new HashMap<>();
+        ItemStack stack1 = new ItemStack(Blocks.LADDER);
+        ItemStack stack2 = new ItemStack(Blocks.OAK_FENCE);
+        ItemStack stack3 = new ItemStack(Blocks.TORCH);
+        ItemStack stack4 = new ItemStack(Blocks.COBBLESTONE);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'stack4' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)